src/actions/index.js

```diff
@@ -0,0 +1,287 @@
+export const CREATE_AUTH = 'CREATE_AUTH';
+export const LOGIN_AUTH_SUCCESS = 'LOGIN_AUTH_SUCCESS';
+export const LOGIN_AUTH_ERROR = 'LOGIN_AUTH_ERROR';
+export const GET_TOKEN_SUCCESS = 'GET_TOKEN_SUCCESS';
+export const GET_TOKEN_ERROR = 'GET_TOKEN_ERROR';
+export const RENEW_ID_TOKEN_SUCCESS = 'RENEW_ID_TOKEN_SUCCESS';
+export const RENEW_ID_TOKEN_ERROR = 'RENEW_ID_TOKEN_ERROR';
+export const DECODE_ID_TOKEN_SUCCESS = 'DECODE_ID_TOKEN_SUCCESS';
+export const DECODE_ID_TOKEN_ERROR = 'DECODE_ID_TOKEN_ERROR';
+export const REFRESH_SESSION_SUCCESS = 'REFRESH_SESSION_SUCCESS';
+export const REFRESH_SESSION_ERROR = 'REFRESH_SESSION_ERROR';
+export const CLOSE_SESSION_SUCCESS = 'CLOSE_SESSION_SUCCESS';
+export const CLOSE_SESSION_ERROR = 'CLOSE_SESSION_ERROR';
+export const LOGOUT_AUTH_SUCCESS = 'LOGOUT_AUTH_SUCCESS';
+export const LOGOUT_AUTH_ERROR = 'LOGOUT_AUTH_ERROR';
+
+const BASE_URL = "https://dev-570863.oktapreview.com";
+const CLIENT_ID = "OaCz3jBQxbaEnsDAFO3A";
+const REDIRECT_URI = "http://localhost:8080/";
+
+const RESPONSE_TYPE = ['id_token', 'token'];
+var CLIENT_SCOPES = ['openid', 'email', 'profile', 'groups'];
+
+
+// CREATE AUTH ACTIONS
+
+/**
+ *  Creates the Okta Authentication binding
+ */
+export function createAuth(uri) {
+  const request = new OktaAuth({
+    url: BASE_URL,
+    clientId : CLIENT_ID,
+    redirectUri: REDIRECT_URI
+  });
+
+  return {
+    type: CREATE_AUTH,
+    payload: request
+  };
+}
+
+// LOGIN ACTIONS
+
+/**
+ *  Uses the Okta AuthSDK to establish a session given
+ *  "username" and "password" 
+ */
+export function loginAuth(user, auth) {
+  return function(dispatch) {
+    auth.signIn({
+      username: user.username,
+      password: user.password
+    })
+    .then((response) => {
+      dispatch(loginAuthSuccess(response));
+    })
+    .catch((error) => {
+      dispatch(loginAuthError(error));
+    });
+  };
+}
+
+function loginAuthSuccess(response) {
+  if(response.status === 'SUCCESS') {
```

Comment:
Btw, re the question of where to put the logic for redirecting - I was reading the redux documentation last night, and it does seem like it's okay to put this logic in an action creator. From what I understand:
- The store describes our entire app state
- A reducer is a **pure** function that has the signature (currentState, action) -> newState. It can be broken up into smaller reducers that interact with different parts of the store object. It is important that this is a pure function (i.e. does not modify state, and does not have side effects or depend on global variables)
- Actions are simple javascript objects that describe what happened in the system, and are sent when we want to update the store
- Action creators dispatch actions to the store. These can have side effects, and be async. It's a tiny comment, but they describe this in the [middleware](http://redux.js.org/docs/advanced/Middleware.html) section - "People use Redux middleware for logging, crash reporting, talking to an asynchronous API, routing, and more." <-- btw, middleware seems like it's a layer inside of the action creator - I call an action creator function, which has this layer to determine what actions to dispatch (and can be asynchronous, have side effects, and route i.e. cancel out of this whole flow).